### ISSUE TITLE
Feature Request: azurerm_container_group : add log analytics workspace settings

### ISSUE THREAD

**Issue by adnang** [terraform-providers/terraform-provider-azurerm#2755]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We should add a log_analytics_workspace settings block to the azurerm_container_group resource which will enable integration with log analytics workspace.

### New or Affected Resource(s)

* azurerm_container_group

### Potential Terraform Configuration

```hcl
resource "azurerm_container_group" "aci-helloworld" {
  name                = "aci-hw"
  location            = "${azurerm_resource_group.aci-rg.location}"
  resource_group_name = "${azurerm_resource_group.aci-rg.name}"
  ip_address_type     = "public"
  dns_name_label      = "aci-label"
  os_type             = "Linux"

  container {
    name   = "hw"
    image  = "seanmckenna/aci-hellofiles"
    cpu    = "0.5"
    memory = "1.5"
    ports  = {
      port     = 80
      protocol = "TCP"
    }
  }

  log_analytics_workspace {
    workspace_id = "00000000-0000-0000-0000-000000000000"
    workspace_key = ""
  }
  
  tags {
    environment = "testing"
  }
}
```

**Comment by metacpp** [comment]
@adnang Thanks for opening this issue. I did a quick investigation on Azure Go SDK side, Our proposed schema is:

```hcl
log_analytics {
  workspace_id = ""
  workspace_key = ""
  log_type = ""
  metadata {
     "test-key": "test-metadata-value"
   }
}
```

**Comment by adnang** [comment]
@metacpp looks good

**Comment by Lucretius** [comment]
@tombuildsstuff looks like this one can be closed

**Comment by metacpp** [comment]
@adnang The PR was merged in v1.23.0, so I am going to close this issue.
